README.md

```diff
@@ -1,10 +1,60 @@
-# Website
+# Project Layout
+
+### The project is broken down into 4 sections
+- Concepts - General Uniswap information or concepts useful for using Uniswap products, such as *Liquidity* and *Fees*
+- Contracts - The Uniswap smart contracts such as *Permit2*, or groupings such as the V3 Contracts
+- SDKs - The Uniswap SDKs such the *v3-sdk*
+- APIs - The Uniswap APIs such the *Subgraph API*
+
+### Each section contains items of that category
+For each item there should be:
+- *Overview*
+- *Guides*
+- *Technical Reference*
+
+## Deep dive into the requirements of each part
+
+### Overview
+This should be an overview of the product.
+It should address points such as:
+
+- What are the high level components of the product
+- What are the high level functionalities
+- Where does the source code of the product live
+- Where does the code artifact live (eg *npm*) and how to integrate with it
+
+A good example is the [V3 Smart Contracts](./docs/contracts/v3/overview.md).
+
+### Guides
+This should contain guides for using the product.
+Each guide must have 3 parts:
+- An introduction
+- A step-by-step walk through of each line of code. The guide should not include code that is not part of the example, but should instead reference or link those pieces.
+- An output or end state that users can test against
+
+A good example is the [V3 SDK Guides](./docs//sdk/v3/guides/01-quick-start.md).
+
+### Technical References
+This should contain the technical reference for the exported interfaces.
+A good example is the [V3 Smart Contracts](./docs/contracts/v3/reference/overview.md).
+
+## Checklist for adding a new product
+Let's walk through an example of adding documentation for new contract, say *Permit2* 
+
+| Question  | Complete |
+| ------------- | ------------- |
+| Did I pick the right section for the product? | In this case, [contracts](./docs/contracts/)  |
+| Did I create the product folder?   | In this case, [yes](./docs/contracts/permit2/)  |
+| Did I introduce any new concepts? | In this case no, but if so add [here](./docs/concepts/) |
+| Did I include an Overview of the product under <product_name>/overview? | Yes, I did add them [here](./docs/contracts/permit2/overview.md) |
+| Did I include Guides of the product under <product_name>/guides? | They should be added [here](./docs/contracts/permit2/guides) |
+| Did I include Technical Reference of the product under <product_name>/reference? | Yes I added them [here](./docs/contracts/permit2/reference) |
+| Did I open a PR using the the [Contributing](./CONTRIBUTING.md) guidelines? | Yes
```

Comment:
I think the checklist style is better here, this feels a bit repetitive and especially odd when the example doesn't cover all parts. I like the idea of an example though - can we add that as a subsection like: Example: `permit2` and then link to the parts added for it?